src/user/dto/searchUsers.dto.ts

```diff
@@ -0,0 +1,19 @@
+import {IsOptional, MinLength, IsPhoneNumber, IsString} from 'class-validator';
+
+export class SearchUsersDto {
+    @IsString()
+    @IsOptional()
+    ids?: [string];
```

Comment:
Have u tested it? Im pretty sure, you would get an error here:)
You need special decorators to check types within arrays.